pkg/agent/agent_test.go

```diff
@@ -708,3 +712,263 @@ func TestRestorePortConfigs(t *testing.T) {
+		t.Run(tt.name, func(t *testing.T) {
+			initializer := &Initializer{
+				networkConfig: &config.NetworkConfig{
+					IPsecConfig: config.IPsecConfig{},
+				},
+			}
+			if tt.expectedErr == "" {
+				defer mockIPsecPSKEnv("key")()
+			}
+
+			err := initializer.readIPSecPSK()
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestGetNodeConfig(t *testing.T) {
+	nodeConfig := &config.NodeConfig{}
+	initializer := &Initializer{
+		nodeConfig: nodeConfig,
+	}
+	assert.Equal(t, nodeConfig, initializer.GetNodeConfig())
+}
+
+func TestWaitForIPSecMonitorDaemon(t *testing.T) {
+	tests := []struct {
+		name        string
+		expectedErr string
+	}{
+		{
+			name:        "IPsec monitor is not running",
+			expectedErr: "IPsec was requested, but the OVS IPsec monitor does not seem to be running",
+		},
+		{
+			name: "IPsec monitor running",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			initializer := &Initializer{}
+			if tt.expectedErr == "" {
+				appFS := afero.NewMemMapFs()
+				err := appFS.MkdirAll("/var/run/openvswitch", 0777)
+				require.NoError(t, err)
+				_, err = appFS.Create("/var/run/openvswitch/ovs-monitor-ipsec.pid")
+				require.NoError(t, err)
+				appFs = appFS
+				defer func() {
+					appFs = afero.NewOsFs()
+				}()
+			}
+
+			err := initializer.waitForIPsecMonitorDaemon()
+			if tt.expectedErr != "" {
+				assert.ErrorContains(t, initializer.waitForIPsecMonitorDaemon(), tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestInitVMLocalConfig(t *testing.T) {
+	ipDevice := &net.Interface{
+		Name: "fakeUplinkInterface",
+	}
+	testNode := &crdv1alpha1.ExternalNode{
+		ObjectMeta: metav1.ObjectMeta{Name: "testNode", Namespace: "external"},
+		Spec: crdv1alpha1.ExternalNodeSpec{
+			Interfaces: []crdv1alpha1.NetworkInterface{
+				{
+					IPs: []string{"192.168.1.2"},
+				},
+			},
+		},
+	}
+
+	tests := []struct {
+		name        string
+		nodeName    string
+		client      *fake.Clientset
+		crdClient   *fakeversioned.Clientset
+		expectedErr string
+	}{
+		{
+			name:      "Finished VM config initialization",
+			nodeName:  "testNode",
+			crdClient: fakeversioned.NewSimpleClientset(testNode),
+		},
+		{
+			name:        "provided external Node unavailable",
+			nodeName:    "testNode",
+			crdClient:   fakeversioned.NewSimpleClientset(),
+			expectedErr: "timed out waiting for the condition",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			stopCh := make(chan struct{})
+			initializer := &Initializer{
+				client:                tt.client,
+				crdClient:             tt.crdClient,
+				ovsBridge:             "br-int",
+				stopCh:                stopCh,
+				networkConfig:         &config.NetworkConfig{},
+				nodeType:              config.ExternalNode,
+				externalNodeNamespace: "external",
+			}
+			close(stopCh)
```

Comment:
Better to move it after 895 and change it to `defer close(stopCh)`

Reply:
If this is deferred then for second test case it keeps on waiting because of the polling mechanism used in the definition of the function tested here.